### ISSUE TITLE
Add a NewSceneObject event to uniformly initialize objects in a scene

### ISSUE THREAD

**Issue by ironfroggy** [ppb/pursuedpybear#473]
This event would be signalled for all objects in the scene at the start of the scene, after SceneStarted, and also fired for any new object when it is added to a scene.

**Comment by pathunstrom** [comment]
The easiest way I can think to implement this, it will need to be delayed. Basically every time you call add, the scene will need to add it to a list of new objects, and the event will get signaled later (probably the next idle? That seems like the most sensible place.

I assume you'd probably would want the inverse as well? ObjectRemovedFromScene or similar?

**Comment by ironfroggy** [comment]
Yeah, probably want a remove event, too.

Can't the add event be added to the event queue immediately? It will won't actually be published until the queue is processed. I think we don't want a frame to occure between adding the object and this event being handled.

**Comment by pathunstrom** [comment]
Nope. `Scene.add()` doesn't have a handle to `signal`, it'll need to wait until it has one.

**Comment by pathunstrom** [comment]
That said, idle events happen when the event queue clears, so a frame _usually_ shouldn't pass in between unless sprites are getting added during `PreRender`.

**Comment by astronouth7303** [comment]
#249 will probably help with this.
